### PR TITLE
Handle SimulateScenario requests

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -399,7 +399,7 @@ func (p *ParticipantImpl) Start() {
 	})
 }
 
-func (p *ParticipantImpl) Close() error {
+func (p *ParticipantImpl) Close(bool) error {
```
Comment (boks1971) [inline pkg/rtc/participant.go:402]: argument is not named and seems not used? Is it for future use?

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -505,6 +505,34 @@ func (r *Room) OnMetadataUpdate(f func(metadata string)) {
 	r.onMetadataUpdate = f
 }
 
+func (r *Room) SimulateScenario(participant types.Participant, simulateScenario *livekit.SimulateScenario) error {
+	switch scenario := simulateScenario.Scenario.(type) {
+	case *livekit.SimulateScenario_SpeakerUpdate:
+		r.Logger.Infow("simulating speaker update", "participant", participant.Identity())
+		go func() {
+			<-time.After(time.Duration(scenario.SpeakerUpdate) * time.Second)
```
Comment (boks1971) [inline pkg/rtc/room.go:513]: Should we think about parameterising scenarios? For example, the duration here?

**Review by boks1971 — APPROVED** [review]
This will be super useful!! A couple of small comments.

File: pkg/rtc/room.go (hunk 2 above)
Comment (davidzhao) [reply pkg/rtc/room.go:513]: Do you mean a separate duration field? And leaving `SpeakerUpdate` as bool?

Comment (boks1971) [reply pkg/rtc/room.go:513]: Sorry, I was reading things wrong. My bad.

File: pkg/rtc/participant.go (hunk 1 above)
Comment (davidzhao) [reply pkg/rtc/participant.go]: oops, good catch. fixed it locally but forgot to push up.

**Pull request merged**